File: 83-remove-dubs-linked-list/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        
        ListNode runner = head;

        while (runner != null && runner.next != null) {

            if (runner.val == runner.next.val) {
                runner.next = runner.next.next;
            } else {
                runner = runner.next;
            }

        }

        return head;
    }
}

File: 83-remove-dubs-linked-list/Solution.py
from typing import Optional


class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    @staticmethod
    def deleteDuplicates(head: Optional[ListNode]) -> Optional[ListNode]:
        
        runner = head

        while runner is not None and runner.next is not None:
            if (runner.val == runner.next.val):
                runner.next = runner.next.next
            else:
                runner = runner.next
        
        return head